src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -3326,13 +3328,15 @@ int sockinfo_tcp::setsockopt(int __level, int __optname,
 		}
 	}
 
-	si_tcp_logdbg("going to OS for setsockopt level %d optname %d", __level, __optname);
-	ret = orig_os_api.setsockopt(m_fd, __level, __optname, __optval, __optlen);
-	BULLSEYE_EXCLUDE_BLOCK_START
-	if (ret) {
-		si_tcp_logdbg("setsockopt failed (ret=%d %m)", ret);
+	if (sys_call) {
```

Comment:
@OphirMunk I think that this logic should be based on our agreement. I see two options here:
1. current user permission is high priority and can suppress VMA_AVOID_SYS_CALLS_RAW_PERM variable.
2. VMA_AVOID_SYS_CALLS_RAW_PERM is high priority and logic does not depend on user permission.

Current PR follows option 2. I think it is obvious for end user (it is direct mode). I prefer to use one.
But option 1 is reasonable too. So we need to have some final conclusion here.